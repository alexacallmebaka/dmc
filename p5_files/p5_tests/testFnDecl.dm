main : () void { }

a : (x:int) int {
 return 5; 
}

b : (x:int) int { 
  a: int = 5; 
  return a;
}

d : (x:int) int {
 return x; 
}

c : (x:int) int {
 return true; 
}

g : (x:int, f:bool) int {
 return f; 
}

x : (y:int, f:bool) int {
 return; 
}

y : (y:int, f:bool) void {
 return 5; 
}