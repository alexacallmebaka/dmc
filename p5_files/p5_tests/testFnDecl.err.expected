FATAL [17,9]-[17,13]: Bad return value
FATAL [21,9]-[21,10]: Bad return value
FATAL [25,2]-[25,8]: Missing return value
FATAL [29,9]-[29,10]: Return with a value in void function
Type Analysis Failed
